tests/integration/test_context_query.py

```diff
@@ -0,0 +1,414 @@
+"""Integration test for context query system in scriptrag analyze.
+
+This test validates that markdown agents can execute context queries to retrieve
+historical data from the database before LLM analysis.
+"""
+
+import json
+import sqlite3
+from pathlib import Path
+
+import pytest
+from typer.testing import CliRunner
+
+from scriptrag.cli.main import app
+from scriptrag.config import set_settings
+
+runner = CliRunner()
+
+
+@pytest.fixture(autouse=True)
+def clean_settings():
+    """Reset settings before and after each test."""
+    set_settings(None)
+    yield
+    set_settings(None)
+
+
+@pytest.fixture
+def multi_scene_screenplay(tmp_path):
+    """Create a screenplay with multiple scenes containing props."""
+    script_path = tmp_path / "props_context_test.fountain"
+    content = """Title: Props Context Test
+Author: Test Suite
+Draft date: 2024-01-01
+
+INT. DETECTIVE'S OFFICE - DAY
+
+The office is cluttered with case files. DETECTIVE JONES (40s) sits at his desk,
+examining a REVOLVER. His BADGE gleams on the desk next to a half-empty
+BOTTLE OF WHISKEY.
+
+JONES
+(to himself)
+This gun... it's the key to everything.
+
+He puts the revolver in his LEATHER HOLSTER and picks up his FEDORA HAT.
+
+INT. CRIME SCENE - LATER
+
+Jones enters the crime scene. The same REVOLVER from his office is now in an
+EVIDENCE BAG on a table. He pulls out his MAGNIFYING GLASS to examine it closer.
+
+A FORENSICS TECH hands him a MANILA ENVELOPE containing PHOTOGRAPHS.
+
+FORENSICS TECH
+The gun matches the one from the Miller case.
+
+Jones nods, putting the photos back in the envelope. He notices a COFFEE MUG
+on the counter - the same distinctive blue mug from his office.
+
+JONES
+Someone's trying to frame me.
+
+INT. INTERROGATION ROOM - NIGHT
+
+Jones sits across from a SUSPECT. The REVOLVER sits in the evidence bag between them.
+Jones' BADGE is on the table. He pulls out a CIGARETTE and LIGHTER.
+
+SUSPECT
+That's not my gun. I've never seen it before.
+
+Jones slides the PHOTOGRAPHS across the table. The same manila envelope from
+earlier sits nearby.
+
+JONES
+Then explain these photos.
+
+The suspect nervously taps his fingers on the metal TABLE. A TWO-WAY MIRROR
+reflects their tense faces.
+
+FADE OUT.
+"""
+    script_path.write_text(content)
```

Comment:
extract into our existing fixtures directory, make sure that props are not capitalized, only CHARACTER names are.